pkg/agent/runner.go

```diff
@@ -356,8 +371,7 @@ func (r *Runner) setStatus(with func(*podStatus)) {
 
 // Run is the main entrypoint to the long-running per-VM pod tasks
 func (r *Runner) Run(ctx context.Context) error {
-	ctx, cancelCtx := context.WithCancel(ctx)
-	defer cancelCtx()
+	defer r.shutdown()
```

Comment:
I would just do `ctx, r.shutdown = context.WithCancel(ctx); defer r.shutdown()`  so that the context passed to run is definitely canceled.

Reply:
Done